README.md

```diff
@@ -19,46 +19,78 @@ In some cases, it is possible to prefetch the static resources of an application
 
 ## Install
 
 ```shell
 npm install -D @o3r/ngx-prefetch
 ```
+OR
+```shell
+ng add @o3r/ngx-prefetch
+```
+> **NOTE:** This library is following the Angular release cycle. For instance, if you are using Angular 13, use a 13.x version of the library:
+>
+> ```shell
+> npm install -D @o3r/ngx-prefetch@13
+> ```
+> OR 
+> ```shell
+> ng add @o3r/ngx-prefetch@13
+> ```
 
-## Builder options
-  - `targetBuild` **Mandatory** The target build where prefetch should be applied. Used for identifying the `outputPath` of the build.
-
-  - `resourceTypes` An object describing the resource types that should be prefetched.
+## Get started
 
-  - `crossorigin` Flag that sets crossorigin attribute on links. If true it will be set for all prefetched resources.
+1. By running the `ng add` command above, the following lines should have been added to the application's `angular.json`:
 
-  - `production` Flag for creating a production (minified) version of the js file or a development one.
+```json
+"generate-prefetch": {
+    "builder": "@o3r/ngx-prefetch:run",
+    "options": {
+        "targetBuild": "app-name:build:production"
+    }
+},
+```
 
-  - `staticsFullPath` By default the prefetched resources are hosted next to the `ngxPrefetch.js` file, on the same server. If it is not the case, you can configure the full path of the resources that will be prefetched. (ex: https://my-web-app.com/path/to/my-app/). It is also possible to set this value by runtime. Instead of setting it in the Builder's options, you can search and replace for `{STATICS_FULL_PATH}` on the server side in order to inject a path.
-## Usage
+> **NOTE:** Additional configuration can be added to the `angular.json` (builder options are described below with an example of full configuration).
 
-[`package.json`]
+2. Then, add a command to the `package.json` to run `generate-prefetch`:
 
 ```json
-    ...
-    "build:prod": "ng build --prod && yarn run generate:prefetch",
-    "generate:prefetch": "yarn app-name:generate-prefetch",
-    ...
+"generate:prefetch": "ng run app-name:generate-prefetch"
 ```
 
-[`angular.json`]
+3. Run the `ng build` command with the `production` configuration. 
+The default configuration of the build command should be `production`, otherwise it should be specified: 
 
 ```json
-"generate-prefetch": {
-    "builder": "@o3r/ngx-prefetch:run",
-    "options": {
-        "targetBuild": "my-app:build:production"
-    }
-},
+...
+"build:prod": "ng build --configuration production",
+...
 ```
 
+4. Run the previously defined `generate:prefetch` command.
+
+## How it works
+
+Please refer to the details on how the ngx-prefetch works [here](docs/HOW_IT_WORKS.md).
+
+## Builder options
+  - `targetBuild` **Mandatory** The target build where prefetch should be applied. Used for identifying the `outputPath` of the build.
+
+  - `resourceTypes` An object describing the resource types that should be prefetched.
```

Comment:
Is there any constraints related to the key of said object? For instance would "img": ["png"] still be valid?
Does it follow a schema?

Reply:
@vscaiceanu-1a

Reply:
@cpaulve-1A  it's just an object. https://github.com/AmadeusITGroup/ngx-prefetch/blob/master/src/schema.json#L9C1-L10C1